### PR TITLE
Refactor to util

### PR DESCRIPTION
**Pull request by devmanhinton**
This refactors the wrapping of EndDash objects to be done from Template.js and Model.js
via util.js. @bglusman @yaymukund

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -74,4 +74,31 @@ exports.get = function(obj, key) {
   } else {
     return obj[key];
   }
+}
+
+var isEndDashCompatible = function(model) {
```
Comment (yaymukund) [inline lib/util.js:79]: Could this be `isEndDashObject`? Seems more natural.

File: lib/util.js [hunk 2]
```diff
@@ -74,4 +74,31 @@ exports.get = function(obj, key) {
...
+                    ', but templates can only be bound to objects.');
+  }
+  return (typeof model.on == 'function' &&
+          typeof model.once == 'function' &&
+          typeof model.set == 'function' &&
+          typeof model.get == 'function'
+         );
+}
+
+exports.toEndDashObject = function(model) {
```
Comment (yaymukund) [inline lib/util.js]: See my old comment here: https://github.com/Amicus/end-dash/pull/86#discussion_r7438976

File: lib/util.js (hunk 1 above)
Comment (bglusman) [inline lib/util.js:79]: disagree, there is no such thing as an "EndDashObject", there's an EndDash duck type... I guess isEndDashType could convey that as well or better than 'compatible' does, but don't think 'object' does.

Comment (bglusman) [inline lib/util.js:79]: oh, sorry, I seem to be confused, hadn't seen EndDashObject stuff above... not sure I like there, but still reading, sorry.

Comment (yaymukund) [inline lib/util.js:79]: `toEndDashModel` was my original suggestion, but @devmanhinton went with `toEndDashObject`. I just want to keep the `to*` and `is*` methods consistent.. don't care too much what we call it. :/

Comment (devmanhinton) [inline lib/util.js:79]: Changed to 'toEndDashCompatibleModel'. Issue resolved

File: lib/util.js (hunk 2 above)
Comment (devmanhinton) [inline lib/util.js]: That comment has no relevance to this line as it stands

Comment (yaymukund) [inline lib/util.js]: Hm, I think it's clearer to have early termination there.

``` js
if (isEndDashCompatible(model)) {
  return model;
}

// do the actual conversion
```

Plus, you don't need to assign stuff to `newModel`. You can just return `new Backbone.Collection` directly.

Plus, there's absolutely no reason to assign `newModel = model`

**EDIT:** My old comment had example code that did all these things:

``` js
var toEndDashModel = function(model) {
  if (isEndDashModel(model)) {
    return model;
  } else if (_.isArray(model)) {
    return new Backbone.Collection(model);
  } else {
    return new Backbone.Model(model);
  }
};
```

Comment (devmanhinton) [inline lib/util.js]: Ahhh. Cool. Yes. That looks better. (That was not in the other comment though :D)

File: lib/util.js (hunk 1 above)
Comment (yaymukund) [inline lib/util.js:79]: `isEndDashCompatible(model)` should now change to `isEndDashCompatibleModel(model)`

**Pull request merged**